client/components/dropdown/index.js

```diff
@@ -15,6 +15,7 @@ const Dropdown = ( { id, valuesMap, title, description, value, updateValue, erro
 				name={ id }
 				value={ value }
 				onChange={ ( event ) => updateValue( event.target.value ) }
+				disabled={ Boolean( disabled ) }
```

Comment:
Because it's not required, so it can be `true`, `false` or `undefined` (which needs to be interpreted as `false`). To use `getDefaultProps`, this component would need to be a `class` component.